tracer/src/Datadog.Trace/ClrProfiler/CallTarget/CallTargetState.cs

```diff
@@ -2,130 +2,130 @@
+public readonly struct CallTargetState
 {
+    private readonly Scope? _previousScope;
+    private readonly Scope? _scope;
+    private readonly object? _state;
+    private readonly DateTimeOffset? _startTime;
+
+    private readonly IReadOnlyDictionary<string, string>? _previousDistributedSpanContext;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+    /// </summary>
+    /// <param name="scope">Scope instance</param>
+    internal CallTargetState(Scope? scope)
+    {
+        _previousScope = null;
+        _scope = scope;
+        _state = null;
+        _startTime = null;
+        _previousDistributedSpanContext = null;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+    /// </summary>
+    /// <param name="scope">Scope instance</param>
+    /// <param name="previousScope">Previous scope instance</param>
+    /// <param name="previousDistributedSpanContext">Previous distributed span context</param>
+    internal CallTargetState(Scope? scope, Scope? previousScope, IReadOnlyDictionary<string, string>? previousDistributedSpanContext)
+    {
+        _previousScope = previousScope;
+        _scope = scope;
+        _state = null;
+        _startTime = null;
+        _previousDistributedSpanContext = previousDistributedSpanContext;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+    /// </summary>
+    /// <param name="scope">Scope instance</param>
+    /// <param name="state">Object state instance</param>
+    internal CallTargetState(Scope? scope, object? state)
+    {
+        _previousScope = null;
+        _scope = scope;
+        _state = state;
+        _startTime = null;
+        _previousDistributedSpanContext = null;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+    /// </summary>
+    /// <param name="scope">Scope instance</param>
+    /// <param name="state">Object state instance</param>
+    /// <param name="startTime">The intended start time of the scope, intended for scopes created in the OnMethodEnd handler</param>
+    internal CallTargetState(Scope? scope, object? state, DateTimeOffset? startTime)
+    {
+        _previousScope = null;
+        _scope = scope;
+        _state = state;
+        _startTime = startTime;
+        _previousDistributedSpanContext = null;
+    }
+
+    internal CallTargetState(Scope? previousScope, IReadOnlyDictionary<string, string>? previousDistributedSpanContext, CallTargetState state)
+    {
+        _previousScope = previousScope;
+        _scope = state._scope;
+        _state = state._state;
+        _startTime = state._startTime;
+        _previousDistributedSpanContext = previousDistributedSpanContext;
+    }
+
+    /// <summary>
+    /// Gets the CallTarget BeginMethod scope
+    /// </summary>
+    internal Scope? Scope => _scope;
```

Comment:
Totally unrelated to this PR, but just wondering, why do we have the explicit fields and not just use read-only Auto-properties? 🤔